crates/biome_js_analyze/src/semantic_analyzers/nursery/use_sorted_classes/any_class_string_like.rs

```diff
@@ -0,0 +1,225 @@
+impl Visitor for StringLiteralInCallExpressionVisitor {
+    type Language = JsLanguage;
+
+    fn visit(
+        &mut self,
+        event: &WalkEvent<SyntaxNode<Self::Language>>,
+        mut ctx: VisitorContext<Self::Language>,
+    ) {
+        let options = get_options_from_analyzer(ctx.options);
+        if options.functions.is_empty() {
+            return;
+        }
+        match event {
+            WalkEvent::Enter(node) => {
+                // When entering a call expression node, track if we are in a target function and reset
+                // in_arguments.
+                if let Some(call_expression) = JsCallExpression::cast_ref(node) {
+                    self.in_target_function =
+                        is_call_expression_of_target_function(&call_expression, &options.functions);
+                    self.in_arguments = false;
+                }
+
+                // When entering a call arguments node, set in_arguments.
+                if JsCallArguments::cast_ref(node).is_some() {
+                    self.in_arguments = true;
+                }
+
+                // When entering a string literal node, and we are in a target function and in arguments, emit.
+                if let Some(string_literal) = JsStringLiteralExpression::cast_ref(node) {
+                    if self.in_target_function && self.in_arguments {
+                        ctx.match_query(AnyClassStringLike::JsStringLiteralExpression(
+                            string_literal,
+                        ));
+                    }
+                }
+            }
+            WalkEvent::Leave(node) => {
+                // When leaving a call arguments node, reset in_arguments.
+                if JsCallArguments::cast_ref(node).is_some() {
+                    self.in_arguments = false;
+                }
+            }
+        }
+    }
+}
+
+// functions (template chunk) visitor
+// ----------------------------------
+
+// Finds class-like template chunks in tagged template calls defined in the options, e.g. tw`classes`.
+// TODO: template chunk visitor
+
+// query
+// -----
+
+declare_node_union! {
+    /// A string literal, JSX string, or template chunk representing a CSS class string.
+    pub AnyClassStringLike = JsStringLiteralExpression | JsxString | JsTemplateChunkElement
+}
+
+impl AnyClassStringLike {
+    /// Returns the value of the string literal, JSX string, or template chunk.
+    pub fn value(&self) -> Option<String> {
+        match self {
+            AnyClassStringLike::JsStringLiteralExpression(string_literal) => {
+                Some(string_literal.inner_string_text().ok()?.to_string())
+            }
+            AnyClassStringLike::JsxString(jsx_string) => {
+                Some(jsx_string.inner_string_text().ok()?.to_string())
+            }
+            AnyClassStringLike::JsTemplateChunkElement(template_chunk) => {
+                Some(template_chunk.to_string())
+            }
+        }
+    }
```

Comment:
We could avoid the String allocation by using TokenText.
```suggestion
    pub fn value(&self) -> SyntaxResult<TokenText> {
        match self {
            Self::JsStringLiteralExpression(string_literal) => {
                string_literal.inner_string_text()
            }
            Self::JsxString(jsx_string) => {
                jsx_string.inner_string_text()            }
            Self::JsTemplateChunkElement(template_chunk) => {
                Ok(template_chunk.token_text_trimmed())
            }
        }
    }
```

Reply:
Last I checked, I wasn't able to obtain a TokenText from the template chunk, but I'll see if your code does the trick.

Reply:
If I am not wrong the chunk should be a token. You can retrieve token texts on every token.

Reply:
done, but had to do it like `template_chunk.template_chunk_token().ok()?.token_text()`, note also the lack of trimming, this is intentional to avoid losing information (whitespace collapsing behavior might change in the future)